### DISCUSSION TITLE
Mic Mute/Telephony question

### DISCUSSION THREAD

**Discussion by ssteejans** [NicoHood/HID#268] [Q&A]
I want to make a small keyboard that I can use to mute/adjust volume and mute my microphone.
I have succesfully implemented the first task using the consumer type but the only microphone mute I see in the usb hid usage table is under Telephony. I have tried to figure out how the code works to define what type of device is plugged in but I am getting confused in the PROGMEM 
`static const uint8_t _hidSingleReportDescriptorConsumer[] PROGMEM = {
	/* Consumer Control (Sound/Media keys) */
	0x05, 0x0C,									/* usage page (consumer device) */
	0x09, 0x01, 								/* usage -- consumer control */
	0xA1, 0x01, 								/* collection (application) */`

Am I on the right track for muting my microphone? How do I implement the Telephony type?
I know I can use the keyboard type to send mute codes for a specific program (teams: ctrl-shft-M) but I have multiple different applications I regularly use.

**Comment by NicoHood** [comment]
I am not sure if microphone controls are part of the consumer/hid protocol. You'd need to check the usb hid spec.

**Reply by ssteejans** [reply]
microphone controls are not part of the consumer protocol. They are included under the telephony section of the usb hid spec.

**Reply by NicoHood** [reply]
So you'd need to create a separate telephony device. You can start with the consumer and adapt the changes. If you got something running, please let me know.

**Comment by tswfi** [comment]
Did you figure this out? I'm trying to do something similar and any data you gathered would be interesting.

**Comment by ukarakaya** [comment]
Hi, in the recent release of HID Usage Tables (V1.3), video conferencing mute microphone feature is added. Below are the documents:
https://usb.org/sites/default/files/hut1_3_0.pdf
https://www.usb.org/sites/default/files/hutrr106-callstatemanagementcontrol_0.pdf

It uses Generic Desktop (0x01) for Mute function and status LED.

I can't implement it by myself but I can help test it.

**Reply by ssteejans** [reply]
I never got around to looking into implementing the Telephony because it didn't quite do what I was looking for. I will look more into this though. I'm curious (and can find no information) on if the call state management has actually been implemented on the OS side though.

**Reply by ukarakaya** [reply]
I could not find any information for OS support. Theoretically, OS should support soon. I'll track updates.
Below is a HID info page from Windows.
https://docs.microsoft.com/en-us/windows-hardware/drivers/hid/hid-transports
